calcom/cal.com
# Migrate legacy api endpoints to tRPC II

## What does this PR do?

- Replace `/api/book/confirm` api endpoint with `viewer.bookings.confirm` 
- Deletes  `/api/availability/day`, `/api/availability/eventtype` , `/api/event-type`
- refs #3573

Notes for reviewer: 
- Issue mention the route to be `viewer.book.confirm`, but I have used `viewer.bookings.confirm`. 
Reason: `viewer.bookings.editLocation` is also part `bookings` router, so **confirm and editlocation** are part of the same thing.
- authorized function moved inside the resolver.
- current User is accessed by context and not made a query to fetch it, have a look at the orderBy of prev and new change. (they are opposite)
- I have not added the equilvalent of the below code in new Change
```
  if (!session?.user?.id) {
    throw new HttpError({ statusCode: 401, message: "Not authenticated" });
  }
```
Reason: This should probably be handled by a middleware, as you would not want to add the same code for every resolver again and again.
 I couldn't find it though.
  
  
  Let me know if any changes are required.
  
  
## Type of change

## How should this be tested?
To confirm, search the endpoints in the codebase to find out it's not used anywhere.
